libraries/AP_HAL_ChibiOS/hwdef/CubeRedPrimary/hwdef-bl.dat

```diff
@@ -0,0 +1,77 @@
+# hw definition file for processing by chibios_hwdef.py
+# for H757
+
+# MCU class and specific type
+MCU STM32H7xx STM32H757xx
+
+# USB setup
+USB_VENDOR 0x2DAE # ONLY FOR USE BY HEX! NOBODY ELSE
+USB_PRODUCT 0x1059
+USB_STRING_MANUFACTURER "CubePilot"
+USB_STRING_PRODUCT "CubeRed-BL"
+
+define CORE_CM7
+define SMPS_PWR
+
+define WATCHDOG_DISABLED
+
+# crystal frequency
+OSCILLATOR_HZ 24000000
+
+# board ID for firmware load
+APJ_BOARD_ID 1069
+
+FLASH_SIZE_KB 2048
+
+# bootloader is installed at zero offset
+FLASH_RESERVE_START_KB 0
+
+# the location where the bootloader will put the firmware
+# the H743 has 128k sectors
+FLASH_BOOTLOADER_LOAD_KB 128
```

Comment:
Any thoughts on making the bootloader bigger on this one?

Reply:
Given that the bootloader fits in 16k, not sure this is an issue. The flash is far more valuable elsewhere

Reply:
It doesn't, or at least shouldn't, fit in 16k. It's 24ish /w CAN. This particular device can afford the flash because it has 32MB of external flash that the program can run from. First step would be to find out how much a basic Ethernet bootloader would take up and then go from there.

Reply:
Currently with master we are running out of internal flash. I have made it so that MAVLink stuff is on external flash, that frees up some internal flash. We need to find other modules we can move to external flash to free up enough space for 256KB bootloader.

Reply:
This would be an exercise opposite to what we did for builds without internal flash @andyp1per , basically we need to find things which don't run too frequently and move them to external flash. Things like chibios and hal drivers will continue to live in internal flash, a lot of the stuff outside of that can be in external flash.